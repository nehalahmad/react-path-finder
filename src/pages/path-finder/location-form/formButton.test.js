import React from "react";
import FormButton from "./formButton";
import { shallow } from "enzyme";

describe("<FormButton />", () => {
  it("submit button renders", () => {
    const SubmitButtonObj = shallow(
      <FormButton buttonType="submit" buttonLabel="Submit" isPrimary />
    );
    expect(SubmitButtonObj).toMatchSnapshot();
    expect(SubmitButtonObj.text()).toEqual("Submit");

    let mockCallBack = jest.fn();
    SubmitButtonObj.simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });

  it("reset button renders", () => {
    const ResetButtonObj = shallow(
      <FormButton buttonType="reset" buttonLabel="Reset" />
    );
    expect(ResetButtonObj).toMatchSnapshot();
    expect(ResetButtonObj.text()).toEqual("Reset");

    let mockCallBack = jest.fn();
    ResetButtonObj.simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });
});
