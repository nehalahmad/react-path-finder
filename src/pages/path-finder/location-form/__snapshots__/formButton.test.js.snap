// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> reset button renders 1`] = `ShallowWrapper {}`;

exports[`<FormButton /> submit button renders 1`] = `ShallowWrapper {}`;
